### PR TITLE
Add NodeGetCapabilities method to CSI Client

### PR DESCRIPTION
**Pull request by amogh09**
### Summary

* Add support for calling `NodeGetCapabilities` method of CSI Spec to CSI Client. This method will be used in upcoming PRs to perform health checks on EBS Managed Daemon.
* Add a CSI Client with a mock generated using gomock.
* Add `DefaultCSIClient()` function to `csiclient` package. This function will come in handy for subcomponents that need to interact with the default EBS CSI Driver such as health checks for EBS CSI Driver. The implementation is based on a new `DefaultSocketFilePath()` function that returns filepath to the default socket file for EBS CSI Driver. This function is only implemented for Linux currently.

### Testing

Tested by calling the method against a running EBS CSI Driver process.
```
# Start a driver process
$ ./ebs-csi-driver --endpoint=unix:///tmp/csi-driver.sock
I1003 20:14:16.677291    9912 driver.go:45] "Driver Information" Driver="csi-driver" Version="v1.0.0"

# Test against the driver
$ go test -v -tags unit -count 1 ./csiclient
=== RUN   TestNode
    csi_client_test.go:26: resp capabilities:<rpc:<type:STAGE_UNSTAGE_VOLUME > > capabilities <rpc:<type:GET_VOLUME_STATS > >
--- PASS: TestNode (0.00s)
``` 

New tests cover the changes:  no (integration tests planned for later)

### Description for the changelog

Add NodeGetCapabilities method to CSI Client

**Does this PR include breaking model changes? If so, Have you added transformation functions?**
  
No

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: ecs-agent/csiclient/csi_client.go [hunk 1]
```diff
@@ -29,6 +31,10 @@ import (
 const (
 	protocol        = "unix"
 	fsTypeBlockName = "block"
+
+	DefaultImageName      = "ebs-csi-driver"
+	DefaultSocketName     = "csi-driver.sock"
+	DefaultSocketHostPath = "/var/run/ecs/"
```
Comment (fierlion) [inline ecs-agent/csiclient/csi_client.go:37]: we might need linux/windows specific versions of at least the default socket host path.  Thoughts?

Comment (amogh09) [reply ecs-agent/csiclient/csi_client.go:37]: Right.... Let me split into Linux/Windows code paths. Good catch!

Comment (amogh09) [reply ecs-agent/csiclient/csi_client.go]: Updated.

**Review by fierlion — DISMISSED** [review]

File: ecs-agent/csiclient/dummy_csiclient.go [hunk 2]
```diff
@@ -1,55 +0,0 @@
-// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
```
Comment (mye956) [inline ecs-agent/csiclient/dummy_csiclient.go:1]: @xxx0624 Just a heads up regarding this change.

File: ecs-agent/csiclient/csi_client.go [hunk 3]
```diff
@@ -60,6 +62,13 @@ func NewCSIClient(socketIn string) csiClient {
 	return csiClient{csiSocket: socketIn}
 }
 
+// Returns a CSI client configured with default settings.
+// The default socket filepath is "/var/run/ecs/ebs-csi-driver/csi-driver.sock".
+func NewDefaultCSIClient() CSIClient {
```
Comment (mye956) [inline ecs-agent/csiclient/csi_client.go:67]: Q: Will this be used in a future PR? It might be worth adding a comment of when to use this will be used.

Comment (mye956) [reply ecs-agent/csiclient/csi_client.go:67]: Synced offline. Thanks!

File: ecs-agent/csiclient/dummy_csiclient.go (hunk 2 above)
Comment (amogh09) [reply ecs-agent/csiclient/dummy_csiclient.go]: Added it back 😅

File: ecs-agent/csiclient/csi_client.go (hunk 3 above)
Comment (amogh09) [reply ecs-agent/csiclient/csi_client.go:67]: Updated the description! Thanks!

**Review by mye956 — DISMISSED** [review]

File: ecs-agent/csiclient/csi_client_linux_test.go [hunk 4]
```diff
@@ -0,0 +1,23 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
```
Comment (mye956) [inline ecs-agent/csiclient/csi_client_linux_test.go:1]: Sorry one more nit, I think we're missing the go build tags here.
```
//go:build linux && unit
// +build linux,unit 
```

Comment (amogh09) [reply ecs-agent/csiclient/csi_client_linux_test.go:4]: Done. Thanks!

**Review by mye956 — APPROVED** [review]

**Review by fierlion — APPROVED** [review]

**Pull request merged**